proposals/lip_define_state_and_state_transitions_of_dpos_module.md

```diff
@@ -0,0 +1,1477 @@
+                                    "fieldNumber": 3
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "snapshots": {
+            "type": "array",
+            "fieldNumber": 3,
+            "items": {
+                "type": "object",
+                "required": [
+                    "roundNumber",
+                    "activeDelegates",
+                    "delegateWeightSnapshot"
+                ],
+                "properties": {
+                    "roundNumber": {
+                        "dataType": "uint32",
+                        "fieldNumber": 1
+                    },
+                    "activeDelegates": {
+                        "type": "array",
+                        "fieldNumber": 2,
+                        "items": { "dataType": "bytes" }
+                    },
+                    "delegateWeightSnapshot": {
+                        "type": "array",
+                        "fieldNumber": 3,
+                        "items": {
+                            "type": "object",
+                            "required": ["delegateAddress", "delegateWeight"],
+                            "properties": {
+                                "delegateAddress": {
+                                    "dataType": "bytes",
+                                    "fieldNumber": 1
+                                },
+                                "delegateWeight": {
+                                    "dataType": "uint64",
+                                    "fieldNumber": 2
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "genesisData": {
+            "type": "object",
+            "fieldNumber": 4,
+            "required": ["initRounds", "initDelegates"],
+            "properties": {
+                "initRounds": {
+                    "dataType": "uint32",
+                    "fieldNumber": 1
+                },
+                "initDelegates": {
+                    "type": "array",
+                    "fieldNumber": 2,
+                    "items": { "dataType": "bytes" }
+                }
+            }
+        }
+    }
+}
+```
+
+
+#### Genesis State Initialization
+
+During the genesis state initialization stage, the following steps are executed. If any step fails, the block is discarded and has no further effect.
+
+Let `genesisBlockAssetBytes` be the `data` bytes included in the block assets for the DPoS module and let `genesisBlockAssetObject` be the deserialization of `genesisBlockAssetBytes` according to the `genesisDPoSStoreSchema` schema, given above.
+
+* Initial checks on the properties of `genesisBlockAssetObject`:
+    * Accross elements of the `validators` array, all `address` values must be unique, all `name` values must also be unique.
```

Comment:
Did you mean `Across` instead of `Accross`?

Reply:
yes